From c7ade6ba0405d92b66903757a7b72850772c229e Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 13 Sep 2017 16:57:16 +0800
Subject: [PATCH 4531/5242] MLK-19413-20 drm/imx: dpu: crtc: Set crtc group id

commit  3145b030c13ce2356f579c2e4854f4b4956de1d5 from
https://source.codeaurora.org/external/imx/linux-imx.git

The DPU CRTC device driver may get the CRTC group id from
the pdata of the device.  Let's cache it in struct dpu_crtc
so that the driver may use it later.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-crtc.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-crtc.c b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
index cce37f1..bd6ccbf 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-crtc.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
@@ -42,6 +42,7 @@ struct dpu_crtc {
 	struct dpu_plane	**plane;
 	unsigned int		hw_plane_num;
 	unsigned int		stream_id;
+	unsigned int		crtc_grp_id;
 	int			vbl_irq;
 	int			safety_shdld_irq;
 	int			content_shdld_irq;
@@ -633,6 +634,7 @@ static int dpu_crtc_init(struct dpu_crtc *dpu_crtc,
 	init_completion(&dpu_crtc->dec_shdld_done);
 
 	dpu_crtc->stream_id = stream_id;
+	dpu_crtc->crtc_grp_id = pdata->di_grp_id;
 	dpu_crtc->hw_plane_num = plane_grp->hw_plane_num;
 
 	dpu_crtc->plane = devm_kcalloc(dev, dpu_crtc->hw_plane_num,
-- 
1.7.9.5

